converters/toZigbee.js

```diff
@@ -838,159 +838,46 @@ const converters = {
-                    value.hue = utils.mapNumberRange(hsv.h % 360, 0, 360, 0, 65535);
-                    command = 'enhancedMoveToHueAndSaturation';
-                } else {
-                    value.hue = utils.mapNumberRange(hsv.h, 0, 360, 0, 254);
-                    command = 'moveToHueAndSaturation';
-                }
-            } else if (value.hasOwnProperty('h')) {
-                newState.color = {h: value.h};
-                const hsv = utils.gammaCorrectHSV(utils.correctHue(value.h, meta), 100, 100);
-                if (utils.getMetaValue(entity, meta.mapped, 'enhancedHue', 'allEqual', true)) {
-                    value.hue = utils.mapNumberRange(hsv.h % 360, 0, 360, 0, 65535);
-                    command = 'enhancedMoveToHue';
-                } else {
-                    value.hue = utils.mapNumberRange(hsv.h, 0, 360, 0, 254);
-                    command = 'moveToHue';
-                }
-            } else if (value.hasOwnProperty('s')) {
-                newState.color = {s: value.s};
-                const hsv = utils.gammaCorrectHSV(360, value.s, 100);
-                value.saturation = utils.mapNumberRange(hsv.s, 0, 100, 0, 254);
-                command = 'moveToSaturation';
-            } else if (value.hasOwnProperty('hue') && value.hasOwnProperty('saturation')) {
-                newState.color = {hue: value.hue, saturation: value.saturation};
-                const hsv = utils.gammaCorrectHSV(utils.correctHue(value.hue, meta), value.saturation, 100);
-                value.saturation = utils.mapNumberRange(hsv.s, 0, 100, 0, 254);
-                if (utils.getMetaValue(entity, meta.mapped, 'enhancedHue', 'allEqual', true)) {
-                    value.hue = utils.mapNumberRange(hsv.h % 360, 0, 360, 0, 65535);
-                    command = 'enhancedMoveToHueAndSaturation';
-                } else {
-                    value.hue = utils.mapNumberRange(hsv.h, 0, 360, 0, 254);
-                    command = 'moveToHueAndSaturation';
-                }
-            } else if (value.hasOwnProperty('hue')) {
-                newState.color = {hue: value.hue};
-                const hsv = utils.gammaCorrectHSV(utils.correctHue(value.hue, meta), 100, 100);
-                if (utils.getMetaValue(entity, meta.mapped, 'enhancedHue', 'allEqual', true)) {
-                    value.hue = utils.mapNumberRange(hsv.h % 360, 0, 360, 0, 65535);
-                    command = 'enhancedMoveToHue';
-                } else {
-                    value.hue = utils.mapNumberRange(hsv.h, 0, 360, 0, 254);
-                    command = 'moveToHue';
+                value.saturation = utils.mapNumberRange(hsvCorrected.s, 0, 100, 0, 254);
+                value.brightness = utils.mapNumberRange(hsvCorrected.v, 0, 100, 0, 254);
+                if (newColor.hasOwnProperty('h') && newColor.hasOwnProperty('s') && newColor.hasOwnProperty('v')) {
```

Comment:
Coupling back to https://github.com/Koenkk/zigbee-herdsman-converters/pull/2453/files#r606783445,

This can then be simplified to just checking for hue/saturation ignoring the brightness component.

https://github.com/Koenkk/zigbee-herdsman-converters/blob/fa2f41c4514d4cd60e4cb2fe2a2cf3a239088c45/converters/toZigbee.js#L1001-L1006

This can then be move outside of the switch statement and placed behind a check for `brightness` key we extracted earlier.